lib/views/pad_button_view.dart

```diff
@@ -0,0 +1,169 @@
+        backgroundPadButtonsColor != Colors.transparent
+            ? Colors.black45
+            : Colors.transparent,
+        backgroundPadButtonsColor != Colors.transparent
+            ? Colors.black12
+            : Colors.transparent));
+    (buttons.asMap().map((index, paddButton) => MapEntry(
+        index,
+        list.add(createPositionedButtons(
+            paddButton, actualSize, index, innerCircleSize)))));
+    return list;
+  }
+
+  Positioned createPositionedButtons(PadButtonItem paddButton,
+      double actualSize, int index, double innerCircleSize) {
+    return Positioned(
+      child: StatefulBuilder(builder: (context, setState) {
+        return GestureDetector(
+          onTap: () {
+            _processGesture(paddButton, Gestures.TAP);
+          },
+          onTapUp: (details) {
+            _processGesture(paddButton, Gestures.TAPUP);
+            Future.delayed(const Duration(milliseconds: 50), () {
+              setState(() => buttonsStateMap[paddButton.index] =
+                  paddButton.backgroundColor);
+            });
+          },
+          onTapDown: (details) {
+            _processGesture(paddButton, Gestures.TAPDOWN);
+
+            setState(() =>
+                buttonsStateMap[paddButton.index] = paddButton.pressedColor);
+          },
+          onTapCancel: () {
+            setState(() =>
+                buttonsStateMap[paddButton.index] = paddButton.backgroundColor);
+          },
+          onLongPress: () {
+            _processGesture(paddButton, Gestures.LONGPRESS);
+          },
+          onLongPressStart: (details) {
+            _processGesture(paddButton, Gestures.LONGPRESSSTART);
+
+            setState(() =>
+                buttonsStateMap[paddButton.index] = paddButton.pressedColor);
+          },
+          onLongPressUp: () {
+            _processGesture(paddButton, Gestures.LONGPRESSUP);
+
+            setState(() =>
+                buttonsStateMap[paddButton.index] = paddButton.backgroundColor);
+          },
+          child: Padding(
+            padding: EdgeInsets.all(buttonsPadding),
+            child: CircleView.padButtonCircle(
+                innerCircleSize,
+                buttonsStateMap[paddButton.index],
+                paddButton.buttonImage,
+                paddButton.buttonText),
+          ),
+        );
+      }),
+      top: _calculatePositionYOfButton(index, innerCircleSize, actualSize),
+      left: _calculatePositionXOfButton(index, innerCircleSize, actualSize),
+    );
+  }
+
+  void _processGesture(PadButtonItem button, Gestures gesture) {
+    if (padButtonPressedCallback != null &&
+        button.supportedGestures.contains(gesture)) {
+      padButtonPressedCallback(button.index, gesture);
+      print("$gesture paddbutton id =  ${[button.index]}");
+    }
+  }
+
+  double _calculatePositionXOfButton(
+      int index, double innerCircleSize, double actualSize) {
+    double degrees = 360 / buttons.length * index;
```

Comment:
What if we pass buttons as zero-length array? Maybe we could add some assert (for example in constructor)